pkg/server/commands/list_objects.go

```diff
@@ -247,13 +247,13 @@ func (q *ListObjectsQuery) evaluate(
 		concurrencyLimiterCh := make(chan struct{}, q.resolveNodeBreadthLimit)
 
 		for res := range reverseExpandResultsChan {
-			if atomic.LoadUint32(objectsFound) >= maxResults {
+			if !(maxResults == 0) && objectsFound.Load() >= maxResults {
```

Comment:
nit: isn't this just `maxResults != 0`?

Reply:
Yes. See, this is why I had written `getAllResults`, LOL

Reply:
I can't do math.

Reply:
I'll try to make this simpler to read in a future PR. For now I am keeping as is, to be consistent with line 307